src/main/java/reactor/netty/instrumentation/zipkin/HttpServerTracing.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2011-2019 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.netty.instrumentation.zipkin;
+
+import brave.Span;
+import brave.Tracing;
+import brave.http.HttpServerHandler;
+import brave.http.HttpTracing;
+import brave.propagation.TraceContext.Extractor;
+import io.netty.handler.codec.http.HttpHeaders;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.server.HttpServerRequest;
+import reactor.netty.http.server.HttpServerResponse;
+import reactor.util.annotation.NonNull;
+
+/**
+ * Tracing adapter for HTTP servers.
+ *
+ * <p></p>
+ *
+ * <code><pre>
+ *   Tracing tracing = Tracing.newBuilder()
+ *         ...
+ *         .build();
+ *   HttpServerTracing serverTracing = HttpServerTracing.create(tracing);
+ *
+ *   DisposableServer c = HttpServer.create()
+ *     .port(0)
+ *     .handle(serverTracing.andThen(myHandler))
+ *     .bindNow();
+ * </pre></code>
+ */
+public class HttpServerTracing {
```

Comment:
Usually, what we do is something like this...

```java
ReactorNettyTracing reactorTracing = ReactorNettyTracing.create(httpTracing);

....
 DisposableServer c = HttpServer.create()
    .port(0)
    .handle(reactorTracing.serverHandler().andThen(myHandler))
    .bindNow();

```

It isn't required, but more conventional as the top-level type (in this case `ReactorNettyTracing`) is supplied by the object graph. If this library is is server-only, it is probably also fine what you've done.

Reply:
I think it isn't necessary, `brave.http.HttpTracing` is a good abstraction for both client and server configurations and customizations, so `HttpServerTracing` does nothing except wrap an existing hander to provide tracing functionality.